src/com/cdd/bao/util/Util.java

```diff
@@ -957,4 +957,55 @@ public static String expandFileHome(String fn)
 		String home = System.getProperty("user.home");
 		return home + fn.substring(1);
 	}
+	
+	/*
+	 *  Returns a measure of string similarity, using the Levenshtein distance. The returned value is basically the number of
+	 *  permutation operations needed to turn one string into the other, for which 0 means identical. Note that the comparison
+	 *  is case sensitive, so for most use cases the inputs should be converted to the same case.
+	 */
+	public static int stringSimilarity(String str1, String str2)
+	{
+		char[] ch1 = str1.toLowerCase().toCharArray();
+		char[] ch2 = str2.toLowerCase().toCharArray();
+		int sz1 = ch1.length, sz2 = ch2.length;
+		if (sz1 == 0) return sz2;
+		if (sz2 == 0) return sz1;
+
+		int cost = ch1[sz1 - 1] == ch2[sz2 - 1] ? 0 : 1;
+		int lev1 = levenshteinDistance(ch1, sz1 - 1, ch2, sz2) + 1;
+		int lev2 = levenshteinDistance(ch1, sz1, ch2, sz2 - 1) + 1;
+		int lev3 = levenshteinDistance(ch1, sz1 - 1, ch2, sz2 - 1) + cost;
```

Comment:
What is this doing? All examples have lev3 being the smallest value. In all test cases lev3 is the smallest value.  I found one case where either lev1 or lev2 is minimum: `abcde` and `abcdes`